### ISSUE TITLE
Can not connect to postgres panic: parse "postgres://semaphore:)": invalid port ":)" after host

### ISSUE THREAD

**Issue by emollusion** [ansible-semaphore/semaphore#745]
Hi!

I have just tried to set up sempahore and I run in to this issue:
```
# semaphore --config /var/opt/semaphore/config.json 
WARN[0000] parse "postgres://semaphore:)": invalid port ":)" after host 

 Have you run `semaphore setup`?
panic: parse "postgres://semaphore:)": invalid port ":)" after host

goroutine 1 [running]:
github.com/ansible-semaphore/semaphore/cli/cmd.createStore(0xc0000be320, 0x700000000998920)
	/go/src/github.com/ansible-semaphore/semaphore/cli/cmd/root.go:120 +0x21a
github.com/ansible-semaphore/semaphore/cli/cmd.runService()
	/go/src/github.com/ansible-semaphore/semaphore/cli/cmd/root.go:55 +0x4f
github.com/ansible-semaphore/semaphore/cli/cmd.glob..func3(0x1a93b60, 0xc00005e520, 0x0, 0x2)
	/go/src/github.com/ansible-semaphore/semaphore/cli/cmd/service.go:17 +0x25
github.com/ansible-semaphore/semaphore/cli/cmd.glob..func2(0x1a93b60, 0xc00005e520, 0x0, 0x2)
	/go/src/github.com/ansible-semaphore/semaphore/cli/cmd/root.go:34 +0x66
github.com/spf13/cobra.(*Command).execute(0x1a93b60, 0xc000020190, 0x2, 0x2, 0x1a93b60, 0xc000020190)
	/go/src/github.com/ansible-semaphore/semaphore/vendor/github.com/spf13/cobra/command.go:860 +0x2c2
github.com/spf13/cobra.(*Command).ExecuteC(0x1a93b60, 0x1a9c7c0, 0xa33a5c, 0x6)
	/go/src/github.com/ansible-semaphore/semaphore/vendor/github.com/spf13/cobra/command.go:974 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
	/go/src/github.com/ansible-semaphore/semaphore/vendor/github.com/spf13/cobra/command.go:902
github.com/ansible-semaphore/semaphore/cli/cmd.Execute()
	/go/src/github.com/ansible-semaphore/semaphore/cli/cmd/root.go:48 +0x135
main.main()
	/go/src/github.com/ansible-semaphore/semaphore/cli/main.go:8 +0x25
```

This is my config:
```
cat /var/opt/semaphore/config.json
{
 	"mysql": {
 		"host": "",
 		"user": "",
 		"pass": "",
 		"name": ""
 	},
 	"bolt": {
 		"host": "",
 		"user": "",
 		"pass": "",
 		"name": ""
 	},
 	"postgres": {
 		"host": "127.0.0.1:5432",
 		"user": "semaphore",
 		"pass": "****",
 		"name": "semaphore"
 	},
 	"dialect": "postgres",
 	"port": "",
 	"interface": "",
 	"tmp_path": "/tmp/semaphore",
 	"cookie_hash": "xCUuRm+8HVpq87vsC8PoxCBKALPQV93DwGg3fzjoBwQ=",
 	"cookie_encryption": "foeZ93CJvPHI+2M5FvfltVLYizFvvkGirSypOIPEutA=",
 	"access_key_encryption": "6A5mT8PkQ8yyfzyWtEJzqcFyYjrYC26FgG+4kg+IAnw=",
 	"email_sender": "",
 	"email_host": "",
 	"email_port": "",
 	"web_host": "https://localhost",
 	"ldap_binddn": "",
 	"ldap_bindpassword": "",
 	"ldap_server": "",
 	"ldap_searchdn": "",
 	"ldap_searchfilter": "",
 	"ldap_mappings": {
 		"dn": "",
 		"mail": "",
 		"uid": "",
 		"cn": ""
 	},
 	"telegram_chat": "",
 	"telegram_token": "",
 	"concurrency_mode": "",
 	"max_parallel_tasks": 0,
 	"email_alert": false,
 	"telegram_alert": false,
 	"ldap_enable": false,
 	"ldap_needtls": false,
 	"ssh_config_path": ""
 }
```
This is the config derived from semaphore --setup

Any ideas of what the problem could be?

Thanks!

**Comment by fiftin** [comment]
Hm, interesting...

Looks like problem in password :)

**Comment by fiftin** [comment]
https://stackoverflow.com/a/23361849/1035334

I will fix the bug ASAP, but currently you can only use password without special characters.

**Comment by fiftin** [comment]
Fixed in https://github.com/ansible-semaphore/semaphore/releases/tag/v2.8.16

**Comment by jbunner** [comment]
Edited: Was able to resolve the issue, but documenting in case others encounter the same error.
The resolution was to append "?sslmode=disable" to the database name in the environment variable.
SEMAPHORE_DB: semaphore?sslmode=disable

I'll share the complete docker-compose.yml file that worked for me.

--
#semaphore template
version: '3.5'
services: 

    semaphore:
        image: ansiblesemaphore/semaphore
        container_name: semaphore
        restart: unless-stopped
        depends_on:
            - postgres
        environment:
            SEMAPHORE_DB_DIALECT: postgres
            SEMAPHORE_DB_PORT: 5432
            SEMAPHORE_DB_USER: semaphore
            SEMAPHORE_DB_PASS: newpassword
            SEMAPHORE_DB_HOST: postgres
            SEMAPHORE_DB: semaphore?sslmode=disable
            SEMAPHORE_PLAYBOOK_PATH: /tmp/semaphore
            SEMAPHORE_ADMIN_PASSWORD: adminpassword
            SEMAPHORE_ADMIN_NAME: admin
            SEMAPHORE_ADMIN: admin
        ports:
            - 80:3000/tcp

    postgres:
        image: postgres:14-alpine
        container_name: postgres
        restart: unless-stopped
        environment:
            PGDATA: /var/lib/postgresql/data    
            POSTGRES_DB: semaphore
            POSTGRES_PASSWORD: newpassword
            POSTGRES_USER: semaphore
        volumes:
            - ./semaphore/db:/var/lib/postgresql/data:rw


----
Related to postgres connections, I was able to connect to postgres database container, but kept receiving errors "pq: SSL is not enabled on the server". 
It appears that for postgres, pq defaults to SSL unless "sslmode=disable" is set in the connection string.
During the init process, if there is a way to set "options" through another environment variable, would probably resolve that issue.

Not concerned about SSL communication between containers, since they are both on the same Docker network and postgres is isolated from external connections. Yes, there is a way to generate SSL certs and enable SSL mode on the postgres Docker container, but a lot of extra complexity if the two containers are isolated anyway.

docker-compose.yml:
environment:
SEMAPHORE_DB_DIALECT: postgres
SEMAPHORE_DB_PORT: 5432
SEMAPHORE_DB_USER: semaphore
SEMAPHORE_DB_PASS: passwordremoved
SEMAPHORE_DB_HOST: postgres
SEMAPHORE_DB: semaphore  


semaphore container output:

semaphore    | Generated configuration:
semaphore    |  {
semaphore    |  "mysql": {
semaphore    |  "host": "",
semaphore    |  "user": "",
semaphore    |  "pass": "",
semaphore    |  "name": "",
semaphore    |  "options": null
semaphore    |  },
semaphore    |  "bolt": {
semaphore    |  "host": "",
semaphore    |  "user": "",
semaphore    |  "pass": "",
semaphore    |  "name": "",
semaphore    |  "options": null
semaphore    |  },
semaphore    |  "postgres": {
semaphore    |  "host": "postgres:5432",
semaphore    |  "user": "semaphore",
semaphore    |  "pass": "passwordremoved",
semaphore    |  "name": "semaphore",
semaphore    |  "options": null
semaphore    |  },
semaphore    |  "dialect": "postgres",

**Comment by fiftin** [comment]
Thank you @jbunner ,

It is a bug in docker image, variable `SEMAPHORE_DB_OPTIONS` missed.

**Comment by fiftin** [comment]
I'm not a specialist in Docker, is it possible to set environment variable in following manner?

```
    environment:
        SEMAPHORE_DB_OPTIONS:
          sslmode: disable
 ```

**Comment by stuartmaxwell** [comment]
This just caught me out. I'm unable to connect to Postgres from Docker with sslmode disabled. It appears this issue was closed without being fixed?

**Comment by fiftin** [comment]
@stuartmaxwell ,

The issue was closed because it was fixed.

Perhaps you have another issue, please provide more details.

**Comment by stuartmaxwell** [comment]
Hi again - sorry for the lack of details. I am following the docs and trying to use docker-compose with a Postgres database. I got stuck with the same issue reported above, that there appears to be no way to disable sslmode when using docker compose. I searched the GitHub repo for the `SEMAPHORE_DB_OPTIONS` string mentioned above but couldn't find it in the source code so assumed that it was never fixed.

I am using the same environment variables mentioned above, i.e. 
``` SEMAPHORE_DB_DIALECT: postgres
SEMAPHORE_DB_PORT: 5432
SEMAPHORE_DB_USER: semaphore
SEMAPHORE_DB_PASS: passwordremoved
SEMAPHORE_DB_HOST: postgres
SEMAPHORE_DB: semaphore```

But I don't think an environment variable exists to disable sslmode.
